File: akuntansi_db/app/Models/SalesHeaderLine.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class SalesHeaderLine extends Model
{
    use HasFactory;
    protected $fillable = [
        'line_no',
        'sh_id',
        'i_id',
        'qty',
        'unit_price',
        'net_amount',
        'note',
    ];
}

File: akuntansi_flut/lib/services/model/request/sales_header_line.dart
class SalesLineRequest {
  String? lineNo;
  String? iId;
  String? qty;
  String? unitPrice;
  String? netAmount;
  String? note;
  String? itemName;

  SalesLineRequest({this.lineNo, this.iId, this.qty, this.unitPrice, this.netAmount, this.note, this.itemName});

  SalesLineRequest.fromJson(Map<String, dynamic> json) {
    lineNo = json['line_no'];
    iId = json['i_id'];
    qty = json['qty'];
    unitPrice = json['unit_price'];
    netAmount = json['net_amount'];
    note = json['note'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['line_no'] = lineNo;
    data['i_id'] = iId;
    data['qty'] = qty;
    data['unit_price'] = unitPrice;
    data['net_amount'] = netAmount;
    data['note'] = note;
    return data;
  }
}
